### PR TITLE
Review of recognition protos

### PR DESCRIPTION
**Pull request by maciekrb**
Improves on previous iteration of recognition services and proto messages. 

* Fixes previous cardinality issues
* It is now possible to list only tags for all descriptors
* Errors are handled via `set_code` and `set_details` on grpc context
* `Descriptor` renamed to `FeatureDescriptor` to avoid clashing with internal proto attributes

@TODO add `onOf` where applies.

### CODE REVIEW

File: vision/recognition.proto [hunk 1]
```diff
@@ -0,0 +1,27 @@
+
+syntax = "proto3";
+
+package admobilize_vision;
+
+// Describes a recognition feature descriptor
+message FeatureDescriptor {
+
```
Comment (arhuaco) [inline vision/recognition.proto]: If you decide to leave a space after every message definition you should be consistent. You can check the other messages and adjust.

File: vision/recognition.proto [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  // A unique identifier for the descriptor according to UUID v4
+  string uuid = 1; 
+
+  // Descriptor format used by extraction algorithm
+  string data_format = 2;
+
+  // The actual recognition descriptor data
+  repeated float data = 3;
+
+  // Arbitrary tags for the descriptor set
```
Comment (arhuaco) [inline vision/recognition.proto]: A longer explanation of tags. ['nelson', 'matrix-io']

File: vision/recognition.proto [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  repeated string tags = 4;
+}
+
+// Container to group multiple FeatureDescriptors
+// makes storage of multiple FeatureDescriptors more
+// convenient
```
Comment (arhuaco) [inline vision/recognition.proto]: More convenient or possible under X restriction? I don't remember the outcome of the discussion we had.

File: vision/recognition_service.proto [hunk 4]
```diff
@@ -37,7 +38,8 @@ service RecognitionService {
   // When the save is successfull, a descriptor list with the computed
```
Comment (arhuaco) [inline vision/recognition_service.proto]: successful

File: vision/recognition_service.proto [hunk 5]
```diff
@@ -37,7 +38,8 @@ service RecognitionService {
...
   // descriptor data and uuid is returned.
   // The message will contain an error if the descriptor can't be stored.
```
Comment (arhuaco) [inline vision/recognition_service.proto]: descriptors

File: vision/recognition.proto (hunk 2 above)
Comment (maciekrb) [inline vision/recognition.proto]: added

File: vision/recognition.proto (hunk 1 above)
Comment (maciekrb) [inline vision/recognition.proto]: added where missing

**Comment by eighteyes** [comment]
Looks good to me, does the uuid here match to the tracking id?

**Comment by maciekrb** [comment]
@eighteyes  No, the `uuid` in this case is just an identifier for the stored feature descriptor, so it's possible to update / delete them individually.

**Pull request merged**